astropy/convolution/convolve.py

```diff
@@ -563,3 +588,118 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0, crop=True,
+psf_pad=False, fft_pad=True , size= 1024  kernelsize= 1024 Mem usage max=   448.0 Mbyte.  Array, kernel size are   8.4 Mbyte and   8.4 Mbyte. Time use was  2.309 s
+psf_pad=False, fft_pad=False, size= 1024  kernelsize= 1024 Mem usage max=     0.0 Mbyte.  Array, kernel size are   8.4 Mbyte and   8.4 Mbyte. Time use was  2.318 s
+psf_pad=True , fft_pad=True , size= 2048  kernelsize=  256 Mem usage max=  2376.9 Mbyte.  Array, kernel size are  33.6 Mbyte and   0.5 Mbyte. Time use was  6.332 s
+psf_pad=True , fft_pad=False, size= 2048  kernelsize=  256 Mem usage max=   721.0 Mbyte.  Array, kernel size are  33.6 Mbyte and   0.5 Mbyte. Time use was  1.388 s
+psf_pad=False, fft_pad=True , size= 2048  kernelsize=  256 Mem usage max=   444.1 Mbyte.  Array, kernel size are  33.6 Mbyte and   0.5 Mbyte. Time use was  1.109 s
+psf_pad=False, fft_pad=False, size= 2048  kernelsize=  256 Mem usage max=     0.0 Mbyte.  Array, kernel size are  33.6 Mbyte and   0.5 Mbyte. Time use was  1.095 s
+psf_pad=True , fft_pad=True , size= 2048  kernelsize=  512 Mem usage max=  2250.7 Mbyte.  Array, kernel size are  33.6 Mbyte and   2.1 Mbyte. Time use was  6.277 s
+psf_pad=True , fft_pad=False, size= 2048  kernelsize=  512 Mem usage max=   906.3 Mbyte.  Array, kernel size are  33.6 Mbyte and   2.1 Mbyte. Time use was  1.682 s
+psf_pad=False, fft_pad=True , size= 2048  kernelsize=  512 Mem usage max=   572.0 Mbyte.  Array, kernel size are  33.6 Mbyte and   2.1 Mbyte. Time use was  1.076 s
+psf_pad=False, fft_pad=False, size= 2048  kernelsize=  512 Mem usage max=    32.0 Mbyte.  Array, kernel size are  33.6 Mbyte and   2.1 Mbyte. Time use was  1.061 s
+psf_pad=True , fft_pad=True , size= 2048  kernelsize= 1024 Mem usage max=  2267.0 Mbyte.  Array, kernel size are  33.6 Mbyte and   8.4 Mbyte. Time use was  6.458 s
+psf_pad=True , fft_pad=False, size= 2048  kernelsize= 1024 Mem usage max=  1269.3 Mbyte.  Array, kernel size are  33.6 Mbyte and   8.4 Mbyte. Time use was  3.467 s
+psf_pad=False, fft_pad=True , size= 2048  kernelsize= 1024 Mem usage max=   320.0 Mbyte.  Array, kernel size are  33.6 Mbyte and   8.4 Mbyte. Time use was  1.069 s
+psf_pad=False, fft_pad=False, size= 2048  kernelsize= 1024 Mem usage max=    32.0 Mbyte.  Array, kernel size are  33.6 Mbyte and   8.4 Mbyte. Time use was  1.071 s
+psf_pad=True , fft_pad=True , size= 2048  kernelsize= 2048 Mem usage max=  2304.7 Mbyte.  Array, kernel size are  33.6 Mbyte and  33.6 Mbyte. Time use was  6.456 s
+psf_pad=True , fft_pad=False, size= 2048  kernelsize= 2048 Mem usage max=  2296.9 Mbyte.  Array, kernel size are  33.6 Mbyte and  33.6 Mbyte. Time use was  6.445 s
+psf_pad=False, fft_pad=True , size= 2048  kernelsize= 2048 Mem usage max=   192.0 Mbyte.  Array, kernel size are  33.6 Mbyte and  33.6 Mbyte. Time use was  1.099 s
+psf_pad=False, fft_pad=False, size= 2048  kernelsize= 2048 Mem usage max=     0.0 Mbyte.  Array, kernel size are  33.6 Mbyte and  33.6 Mbyte. Time use was  1.091 s
+"""
```

Comment:
And wherever it winds up it would be good to include some relevant specs about the system this profiling was done on.